src/Uno.UI/UI/Xaml/Controls/ListViewBase/ListViewBase.cs

```diff
@@ -101,7 +103,13 @@ private void OnSelectedItemsCollectionChanged(object sender, NotifyCollectionCha
 			try
 			{
 				_modifyingSelectionInternally = true;
-				SelectedItem = SelectedItems.Where(item => items.Contains(item)).FirstOrDefault();
+
+				var itemIndex = SelectedItems.Select(item => (int?)items.IndexOf(item)).FirstOrDefault(index => index > -1);
```

Comment:
Not sure if you added RuntimeTests for the changes you made here, but it'd be good to have some :)

Reply:
This is actually covered by some of the tests in `Uno.UI.Tests`, which helped uncover a bug I introduced this way :-D .